Add more validations

jameswestman opened the pull request:
- Iterate the file tree and check for any executables or shared libraries, and make sure the architecture matches the ref. This uses GLib's [g_content_type_guess](https://docs.gtk.org/gio/func.content_type_guess.html) to find executables and shared libraries and the [elf](https://crates.io/crates/elf) crate to get the target architecture.
- Make sure the screenshots in the appstream catalog file are mirrored and that the files exist in the screenshots/{arch} branch

Also added a validate subcommand to make it easier to test and debug locally.

barthalion commented on src/review/review_files.rs:
```diff
@@ -0,0 +1,108 @@
+
+    let expected_arch = arch_from_ref(refstring);
+    let expected_code = match expected_arch.as_str() {
+        "x86_64" => Some(EM_X86_64),
```
I assume this might not be always valid -- there are apps shipping i386/i686/x86 binaries in x86_64 ref.

jameswestman replied:
I'm not quite sure what the full list of codes should be, but I've added EM_386. It's only a warning anyway.

jameswestman commented:
I refactored the code a bit to make it easier to test. I also added tests that build some manifests and validate them to make sure the output matches what is expected.
